### PR TITLE
VACOLS Integration | Create video master records

### PR DESCRIPTION
**Pull request by oscarAtNava**
Connects #{5450}

### Description
Add ability to create hearing schedule slots in VACOLS. These are scheduling slots with no appeals associated to them.

### Acceptance Criteria 
- [ ] Code compiles correctly
- [ ] Create test case in hearing spec goes green

### Testing Plan
1. Test using Rails console
    a. Store a user in RequestStore : 
        `RequestStore[:current_user] = User.find_by(css_id: "BVASCASPER1")`

    b. Create a scheduled hearing
         `Hearing.create_unassigned_hearing(hearing_type: "C", hearing_date: '29-May-2018', room: "1")`
    c. Output should look something like below, where hearing_pkseq is generated by db trigger:

       ` #<VACOLS::CaseHearing:0x007feea3be1830
       hearing_pkseq: 404113,
       hearing_type: "C",
       folder_nr: nil,
       hearing_date: Tue, 29 May 2018 00:00:00 UTC +00:00,
       hearing_disp: nil,
       board_member: nil,
       notes1: nil,
       team: nil,
       room: "1",
       rep_state: nil, 
       ...
       `

### CODE REVIEW

File: spec/models/hearing_spec.rb [hunk 1]
```diff
@@ -315,4 +315,34 @@
...
+
+  context "#create" do
+    let(:hearing_hash) do
+      { hearing_type: "C",
+        hearing_date: 1.day.ago,
+        representative: "RO89",
+        representative_name: "General Counsel" }
+    end
+
+    let(:hearing) do
```
Comment (sharonwarner) [inline spec/models/hearing_spec.rb]: Why do we need to set this? Maybe we should make this user match the role we'll give the hearing coordinators, "Hearing Schedule"?

File: spec/feature/reader/reader_spec.rb [hunk 2]
```diff
@@ -111,6 +111,14 @@ def add_comment(text)
     FeatureToggle.enable!(:search)
     FeatureToggle.enable!(:test_facols)
     Time.zone = "America/New_York"
+
+    RequestStore[:current_user] = OpenStruct.new(
+      ip_address: "127.0.0.1",
+      station_id: "397",
+      css_id: "AMCDLUCA",
+      roles: ["User", "Manage Claims Establishme", "Establish Claim"],
```
Comment (sharonwarner) [inline spec/feature/reader/reader_spec.rb]: Same with this, not sure if the dispatch permissions make sense.

File: lib/fakes/hearing_repository.rb [hunk 3]
```diff
@@ -33,6 +33,12 @@ def self.update_vacols_hearing!(vacols_record, hearing_info)
     hearing.assign_from_vacols(hearing_info)
   end
 
+  def self.create_vacols_hearing!(hearing_info)
```
Comment (sharonwarner) [inline lib/fakes/hearing_repository.rb]: Now that we have the FACOLS environment, let's get rid of all the generators/fakes. As long as this functionality works in the development environment, we don't need to worry about the stubbed environment.

File: spec/models/hearing_spec.rb (hunk 1 above)
Comment (oscarAtNava) [reply spec/models/hearing_spec.rb]: We need it so the tests can pass when setting mduser in the DB. See CaseHearing.current_user_css_id. Agreed with next comment to put better values for user mocked.

File: app/models/hearing.rb [hunk 4]
```diff
@@ -71,6 +71,12 @@ def update(hearing_hash)
     end
   end
 
+  def self.create_unassigned_hearing(hearing_hash)
+    transaction do
```
[the remark this answers is not in the document]
Comment (oscarAtNava) [reply app/models/hearing.rb]: I changed the transaction for update as you suggested. For the new create method I removed all transaction code as the only model being updated is CaseHearing and if there is a failure ActiveRecord will rollback automatically.

File: app/models/vacols/case_hearing.rb [hunk 5]
```diff
@@ -110,10 +128,16 @@ def update_hearing!(hearing_info)
     MetricsService.record("VACOLS: update_hearing! #{hearing_pkseq}",
                           service: :vacols,
                           name: "update_hearing") do
-      update(attrs.merge(mdtime: VacolsHelper.local_time_with_utc_timezone))
+      update(attrs.merge(mduser: current_user_css_id, mdtime: VacolsHelper.local_time_with_utc_timezone))
```
[the remark this answers is not in the document]
Comment (oscarAtNava) [reply app/models/vacols/case_hearing.rb]: Removed mduser and mdtime from the create operation.

[the remark this answers is not in the document]
Comment (oscarAtNava) [reply app/models/vacols/case_hearing.rb]: I need a bit more guidance on how to get reference to a user object within CaseHearing.

[the remark this answers is not in the document]
Comment (oscarAtNava) [reply app/models/vacols/case_hearing.rb]: Mocked user and staff records successfully to properly set up the create test. In a subsequent PR I will add more create scenarios as requirements solidify all the use cases.

File: app/models/vacols/case_hearing.rb [hunk 6]
```diff
@@ -73,6 +77,18 @@ def load_hearing(pkseq)
       select_hearings.find_by(hearing_pkseq: pkseq)
     end
 
+    def create_hearing!(hearing_info)
+      attrs = hearing_info.each_with_object({}) { |(k, v), result| result[COLUMN_NAMES[k]] = v }
+      attrs.except!(nil)
+      MetricsService.record("VACOLS: create_hearing!",
+                            service: :vacols,
+                            name: "create_hearing") do
+        create(attrs.merge(addtime: VacolsHelper.local_time_with_utc_timezone,
```
Comment (aroltsch) [inline app/models/vacols/case_hearing.rb]: Are any of the fields required? Does it make sense to add ActiveRecord validations?

**Comment by aroltsch** [comment]
@oscarAtNava could you add a snippet how Postman request should look like please

**Pull request merged**